### PR TITLE
Set reasonable defaults for hosted OCP cluster parameters

### PR DESCRIPTION
**Pull request by dahorak**
Set reasonable defaults for follwoing hosted OCP cluster parameters:
- `cpu_cores_per_hosted_cluster`
- `memory_per_hosted_cluster`
- `nodepool_replicas`

The defaults are set the same as in [create_kubevirt_ocp_cluster()](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/helpers/hypershift_base.py#L340-L350).

### CODE REVIEW

File: ocs_ci/deployment/hosted_cluster.py [hunk 1]
```diff
@@ -411,16 +411,21 @@ def deploy_ocp(
         )
 
         ocp_version = config.ENV_DATA["clusters"].get(self.name).get("ocp_version")
+        # use default value 6 for cpu_cores_per_hosted_cluster as used in create_kubevirt_ocp_cluster()
```
[the remark this answers is not in the document]
Comment (dahorak) [reply ocs_ci/deployment/hosted_cluster.py:414]: Updated in [be64fc4](https://github.com/red-hat-storage/ocs-ci/pull/10841/commits/be64fc427b181f5a76a70f2f8c293bab206cbb75).

**Review by fbalak — COMMENTED** [review]
Those parameters are affecting only hosted clusters and are not used in other platforms (so the defaults don't need to be changed for other platforms afaiu). In that case those defaults can be stored in standard place for default config values: https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/framework/conf/default_config.yaml

**Comment by dahorak** [comment]
> Those parameters are affecting only hosted clusters and are not used in other platforms (so the defaults don't need to be changed for other platforms afaiu). In that case those defaults can be stored in standard place for default config values: https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/framework/conf/default_config.yaml

Those parameters are configured per cluster in: `ENV_DATA["clusters"][<cluster_name>][...]`, so it is not possible to configure it directly in `default_config.yaml`. But it might be worth to set it in constants, @fbalak WDYT?

**Comment by fbalak** [comment]
I see. In that case I suggest to store those values in https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/defaults.py.

**Comment by dahorak** [comment]
* rebased and resolved conflicts
* moved default values to `defaults` module

**Review by DanielOsypenko — APPROVED** [review]

**Comment by dahorak** [comment]
Verification job https://url.corp.redhat.com/121a066

Definition of second cluster `j038-c2bai-t1` was without following parameters `cpu_cores_per_hosted_cluster`, `memory_per_hosted_cluster`, `nodepool_replicas`:

```
    j038-c2bai-t1:
      ocp_version: '4.16'
      cluster_conf: conf/deployment/fusion_hci_pc/client_bm_upi_1az_rhcos_nvme_2w.yaml
      yaml_text_config: |
        ENV_DATA:
          cluster_namespace: "openshift-storage"
      odf_version: '4.17'
      ocs_registry_image: ''
      skip_ocp_deployment: false
      hosted_cluster_path: ~/clusters/j038-c2bai-t1/openshift-cluster-dir/
      setup_storage_client: false
```
So the defaults were properly used:
```
11:39:47 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: /home/jenkins/bin/hypershift create cluster kubevirt --name j038-c2bai-t1 --release-image quay.io/openshift-release-dev/ocp-release:4.16.9-x86_64 --node-pool-replicas 2 --memory 12Gi --cores 6 --root-volume-size 40 --pull-secret /home/jenkins/workspace/qe-deploy-ocs-cluster/ocs-ci/data/pull-secret --image-content-sources /tmp/icsp_mirrors-usbxkwpc --annotations 'hypershift.openshift.io/skip-release-image-validation=true' --olm-catalog-placement Guest
```

**Pull request merged**